src/pubkey.cpp

```diff
@@ -166,6 +167,23 @@ static int ecdsa_signature_parse_der_lax(const secp256k1_context* ctx, secp256k1
     return 1;
 }
 
+bool XOnlyPubKey::VerifySchnorr(const uint256 &hash, const std::vector<unsigned char>& sigbytes) const {
+    if (sigbytes.size() != 64) return false;
```

Comment:
nit: how about `!= sizeof(((secp256k1_schnorrsig){0}).data)` or similar?

Reply:
In the end, all of this is probably personal taste but I think there is a reason why a lot of those magic numbers are just literals. This number will never need to change.
 `sizeof(((secp256k1_schnorrsig){0}).data)` just makes the code harder to read and review (because you need to look up the definition of `secp256k1_schnorrsig`, and it would be a little bit like having a constant `BITS_IN_UINT256 = 256`.

Reply:
The notion is that by tying one constant to a formal description, we can make explicit and enforced a connection that would otherwise be implicit and unenforced. IMO, that such a thing can't be violated is always better than that it won't be violated.

There are ways to do that without making the code markedly more complex, such as offloading it to a constant + `static_assert`.

Reply:
I agree with real-or-random.

secp256k1_schnorrsig should be "functionally opaque"-- like the other secp256k1_ types it isn't actually an opaque struct *only* so it can be allocated on the stack. Library users should not be messing around with the internals of these types, and the library API makes *no* promise that the internal representation won't change (and for some of them it's even currently different on different platforms-- although not this one).

If the representation of  secp256k1_schnorrsig were to change, signatures with sizes other than 64 bytes wouldn't suddenly become valid in Bitcoin.

If you were talking about a '64' who's purposes was to allocate memory to store a secp256k1_schnorrsig then your reflex to interrogate it with sizeof would be well placed-- but that isn't the case here.  The data field in a secp256k1_schnorrsig could change length pretty arbitrarily (for example, storing R decompressed) and this line wouldn't change).

Reply:
I agree as well. The size of serialized signatures is not inherently related to the in-memory representation of secp256k1_schnorrsig.

The BIP says something about sizes 64 and 65. The code does the exact same thing. You could introduce a constant, but I think that's just distracting. You'd need to go look up the constant to verify the code is correct, while now you can just directly compare it with the BIP.